src/NuGet.Core/NuGet.Commands/RestoreCommand/RestoreRequest.cs

```diff
@@ -153,7 +153,12 @@ public LockFile ExistingLockFile
         /// <summary>
         /// MSBuildProjectExtensionsPath
         /// </summary>
-        public string MSBuildProjectExtensionsPath { get; set; }
```

Comment:
I'm really not sure why these changes are being made. 

This keeps increasing the scope of the PR.

Reply:
I thought it was a leftover from when you guys combined the `BaseIntermediateOutputPath`, `MSBuildProjectExtensionsPath` into `RestoreOutputPath`. I stumbled upon these when I was trying to fix some tests. So, I corrected it!! may be?

Can I separate those into a new PR, then?

Reply:
Stuff in NuGet.Core is part of a public API. 
We're a tool first, but there are customer that use our packages as libraries, so we've strayed away from making refactoring changes.

Reply:
We added this doc recently, https://github.com/NuGet/NuGet.Client/blob/dev/docs/nuget-sdk.md

Reply:
Looks like Project Json uses it. So, I have reverted this particular change. Thanks for the heads up!!